aci/resource_aci_l3extvirtuallifp.go

```diff
@@ -209,7 +210,31 @@ func resourceAciVirtualLogicalInterfaceProfile() *schema.Resource {
 				},
 			},
 		}),
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, m interface{}) error {
+			configOld, configNew := diff.GetChange("relation_l3ext_rs_dyn_path_att")
+			updatedConfig := compareConfigRsDynPathAttEncapBetweenUnknownEmpty(configOld.(*schema.Set).List(), configNew.(*schema.Set).List())
+			diff.SetNew("relation_l3ext_rs_dyn_path_att", updatedConfig)
+			return nil
+		},
+	}
+}
+
+func compareConfigRsDynPathAttEncapBetweenUnknownEmpty(old []interface{}, new []interface{}) []interface{} {
+	for _, old_map := range old {
+		for _, new_map := range new {
+			if old_map.(map[string]interface{})["tdn"] == new_map.(map[string]interface{})["tdn"] {
+				for key, value_old := range old_map.(map[string]interface{}) {
+					if key == "encap" {
+						if value_old == "unknown" && new_map.(map[string]interface{})[key] == "" {
+							new_map.(map[string]interface{})["encap"] = "unknown"
+							break
+						}
+					}
+				}
```

Comment:
Do we need this loop? Can't we get the encap value from old_map, and new_map directly?
```suggestion
				if old_map.(map[string]interface{})["encap"] == "unknown" && new_map.(map[string]interface{})["encap"] == "" {
					new_map.(map[string]interface{})["encap"] = "unknown"
					break
				}
```